modules/seqexec/server/src/main/scala/seqexec/server/SequenceGen.scala

```diff
@@ -52,28 +52,27 @@ object SequenceGen {
   }
 
   object StepGen {
-    def generate[F[_]](stepGen: StepGen[F], ctx: HeaderExtraData): EngineStep[F] = stepGen match {
-      case p: PendingStepGen[F]          => EngineStep.init[F](stepGen.id, p.generator.generate(ctx))
-      case SkippedStepGen(id, _, _)      => EngineStep.skippedL.set(true)(EngineStep.init[F](id, Nil))
-      case CompletedStepGen(id, _, _, _) => EngineStep.init[F](id, Nil)
-    }
+    def generate[F[_]](stepGen: StepGen[F], systemOverrides: SystemOverrides, ctx: HeaderExtraData): EngineStep[F] =
+      stepGen match {
+        case p: PendingStepGen[F]          => EngineStep.init[F](stepGen.id, p.generator.generate(ctx, systemOverrides))
+        case SkippedStepGen(id, _, _)      => EngineStep.skippedL.set(true)(EngineStep.init[F](id, Nil))
+        case CompletedStepGen(id, _, _, _) => EngineStep.init[F](id, Nil)
+      }
   }
 
-  final case class StepActionsGen[F[_]](pre: List[ParallelActions[F]],
-                                  configs: Map[Resource, Action[F]],
-                                  post: HeaderExtraData => List[ParallelActions[F]]) {
-    def generate(ctx: HeaderExtraData): List[ParallelActions[F]] =
-      pre ++
-        NonEmptyList.fromList(configs.values.toList).toList ++
-        post(ctx)
+  final case class StepActionsGen[F[_]](configs: Map[Resource, SystemOverrides => Action[F]],
+                                        post: (HeaderExtraData, SystemOverrides) => List[ParallelActions[F]]) {
```

Comment:
The changes to `StepActionsGen` are the core of the systems override feature. Making the `Action`s dependent on a `SystemOverrides` allows to reapply it when it changes.